src/shogun/classifier/LPBoost.cpp

```diff
@@ -100,7 +100,13 @@ float64_t CLPBoost::find_max_violator(int32_t& max_dim)
 
 bool CLPBoost::train_machine(CFeatures* data)
 {
-	ASSERT(m_labels)
+	train_machine(data, m_labels);
+	return true;
+}
+
+virtual void CLPBoost::train_machine(CFeatures* features, CLabels* labels);
+{
+	ASSERT(labels)
```

Comment:
I think handling those asserts happens (should happen?) in the base class?

Reply:
Agree, it should be in the base class. I will check if it is possible for now.